src/EdgeAnalysis.jl

```diff
@@ -42,31 +42,50 @@ end
 """
 Generate appropriate edge Simulation/SystemSimulation object
 """
-function getsim(inte,react,coreedgedomain,inters,p,coretoedgespcmap)
-    ycoreedge = getycoreedge(inte.u,coretoedgespcmap,coreedgedomain.indexes[end])
-    sol = build_solution(react.ode,inte.alg,[0.0,inte.t],[ycoreedge,ycoreedge])
+function getsim(inte,react,edgereact,coreedgedomain,inters,p,coretoedgespcmap)
+    ycoreedge = getycoreedge(inte.u,coretoedgespcmap,coreedgedomain.indexes[end],react,coreedgedomain)
+    sol = build_solution(edgereact.ode,inte.alg,[0.0,inte.t],[ycoreedge,ycoreedge])
     sim = Simulation(sol,coreedgedomain,inters,p)
     return sim
 end
 
 """
 Generate appropriate edge Simulation/SystemSimulation object
 """
-function getsim(inte,react,coreedgedomains::Tuple,inters,p,coretoedgespcmap)
-    ycoreedge = getycoreedge(inte.u,coretoedgespcmap,coreedgedomains[end].indexes[end])
-    sol = build_solution(react.ode,inte.alg,[0.0,inte.t],[ycoreedge,ycoreedge])
+function getsim(inte,react,edgereact,coreedgedomains::Tuple,inters,p,coretoedgespcmap)
+    ycoreedge = getycoreedge(inte.u,coretoedgespcmap,coreedgedomains[end].indexes[end],react,coreedgedomains)
+    sol = build_solution(edgereact.ode,inte.alg,[0.0,inte.t],[ycoreedge,ycoreedge])
     ssys = SystemSimulation(sol,coreedgedomains,inters,p)
     return ssys
 end
 
 """
 Generate a state vector appropriate for both the edge and core from the core state vector
 """
-function getycoreedge(y,coretoedgespcmap,numedgespc)
-    ycoreedge = zeros(numedgespc)
+function getycoreedge(y,coretoedgespcmap,edgelen,react,coreedgedomains::Tuple)
+    ycoreedge = zeros(edgelen)
     for (coreind,edgeind) in coretoedgespcmap
         ycoreedge[edgeind] = y[coreind]
     end
+    for (i,domain) in enumerate(coreedgedomains)
+        for j in 3:length(domain.indexes)
+            ycoreedge[domain.indexes[j]] = y[react.domain[i].indexes[j]]
+        end
+    end
+    return ycoreedge
+end
+
+"""
+Generate a state vector appropriate for both the edge and core from the core state vector
+"""
+function getycoreedge(y,coretoedgespcmap,edgelen,react,coreedgedomain)
+    ycoreedge = zeros(edgelen)
+    for (coreind,edgeind) in coretoedgespcmap
+        ycoreedge[edgeind] = y[coreind]
+    end
+    for j in 3:length(coreedgedomain.indexes)
+        ycoreedge[coreedgedomain.indexes[j]] = y[react.domain.indexes[j]]
```

Comment:
Could you add @inbounds to several places in this PR that uses indexing?

Reply:
Done